### PR TITLE
DOP-4758: remove flag for rstspec

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4758

### Notes
- Makefiles should point at the env var ([specified in Dockerfile](https://github.com/mongodb/docs-worker-pool/pull/1057/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R25)) or fallback to the main branch for latest rstspec

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Comment by seungpark** [comment]
Removed the DO NOT MERGE label as it will safely fall back to `main` branch without the env flags - the rstspec consolidation has [been merged to main](https://github.com/mongodb/snooty-parser/pull/607)

**Comment by i80and** [comment]
Is there a reason we don't want to simply use the built-in rstspec? Under what circumstances would we want to build with a different parser version's rstspec?

**Comment by seungpark** [comment]
> Is there a reason we don't want to simply use the built-in rstspec? Under what circumstances would we want to build with a different parser version's rstspec?

discussed offline! updated to remove the flag completely as it will be loaded from the same version if not specified

File: makefiles/shared.mk [hunk 1]
```diff
@@ -14,8 +14,6 @@ PATCH_ID=$(shell if test -f "${PATCH_FILE}"; then git patch-id < ${PATCH_FILE} |
 PATCH_CLAUSE=$(shell if [ ! -z "${PATCH_ID}" ]; then echo --patch "${PATCH_ID}"; fi)
 
 BUNDLE_PATH=${REPO_DIR}/bundle.zip
-RSTSPEC_FLAG=--rstspec=https://raw.githubusercontent.com/mongodb/snooty-parser/latest/snooty/rstspec.toml
```
Comment (i80and) [inline makefiles/shared.mk:17]: We should also remove the `${RSTSPEC_FLAG}` references in makefiles that override the `next-gen-parse` rule. I count 12 of them :(

* `Makefile.docs-atlas-cli`
* `Makefile.docs-atlas-operator`
* `Makefile.docs-landing`
* `Makefile.docs-laravel`
* `Makefile.mongocli`
* `Makefile.mongodb-internal-base`
* `Makefile.docs-mongodb-internal`
* `Makefile.mongoid`
* `Makefile.docs-php-library`
* `Makefile.docs-ruby`
* `Makefile.docs`
* `Makefile.docs-stage`

Comment (seungpark) [reply makefiles/shared.mk:17]: great catch! removed from these makefiles

**Review by i80and — APPROVED** [review]
Good riddance -- this was always such a pain!

**Pull request merged**